lib/toml/string.rb

```diff
@@ -10,26 +10,45 @@ def value
     # Replace the unicode escaped characters with the corresponding character
     # e.g. \u03B4 => ?
     def self.decode_unicode(str)
-      str.gsub(/([^\\](?:\\\\)*\\u[\da-f]{4})/i) do |m|
-        m[0...-6] + [m[-4..-1].to_i(16)].pack('U')
+      str.gsub(/\\u([\da-f]{4,6})/i) do |m|
```

Comment:
According to the specification https://github.com/toml-lang/toml#string \uXXXX and \UXXXXXXXX (with 8 chars not 6 - my mistake in the reported issue) are two different escape sequences. \uXXXXXXXX and \UXXXX are invalid.